infra/resources/prod/main.tf

```
@@ -3,7 +3,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.116"
+      version = ">= 4.0.0"
```

```suggestion
      version = "~> 4.0"
```
To avoid upgrades to v5